src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/editmode/VisualEditModePageObject.java

```diff
@@ -289,144 +258,55 @@ public void removeComponent(Components component) {
-    jsActions.scrollToElementInModal(categoryInput,modalElement);
-    wait.forElementVisible(categoryInput);
-    categoryInput.sendKeys(categoryName);
-    PageObjectLogging.log("typeCategoryName", categoryName + " typed", true);
-  }
-
-  public void triggerCategorySuggestions() {
-    int timeout = 0;
-    pressDownArrow(categoryInput);
-    JavascriptExecutor js = (JavascriptExecutor) driver;
-    String returned = (String) js.executeScript("return $('ul.ui-autocomplete li').text()");
-    while (returned.isEmpty() && timeout <= 5000) {
-      try {
-        Thread.sleep(500);
-        timeout += 500;
-      } catch (InterruptedException e) {
-        PageObjectLogging
-            .log("triggerCategorySuggestions", "Interrupted Exception occurred", false);
+    editorFrame.frameScope(() -> {
+      switch (component) {
+        case PHOTO:
+          wait.forElementNotPresent(IMAGE_BY);
+          break;
+        case GALLERY:
+          wait.forElementNotPresent(galleryBy);
+          break;
+        case SLIDESHOW:
+          wait.forElementNotPresent(slideshowBy);
+          break;
+        case SLIDER:
+          wait.forElementNotPresent(sliderBy);
+          break;
+        case VIDEO:
+          wait.forElementNotPresent(videoBy);
+          break;
+        default:
+          PageObjectLogging.log("verifyComponentRemoved", "Invalid component: " + component.name() + " selected", false);
+          break;
       }
-      pressDownArrow(categoryInput);
-      returned = (String) js.executeScript("return $('ul.ui-autocomplete li').text()");
-    }
+    });
+
+    PageObjectLogging.log("verifyGalleryRemoved", "Click on 'remove button' on gallery", true);
   }
 
-  public void submitCategory(){
-    int categoryListSizeBeforeSubmission = categoryList.size();
-    new Actions(driver).sendKeys(categoryInput, Keys.ARROW_DOWN)
-        .sendKeys(categoryInput, Keys.ARROW_DOWN).sendKeys(categoryInput, Keys.ENTER).perform();
-    PageObjectLogging.log("submitCategory", "category submitted", true);
+  public void removePortableInfobox() {
+    verifyComponent(portableInfoboxTransclusion);
+    driver.executeScript("$('.cke_wysiwyg_frame').contents().find('.placeholder-double-brackets .portable-infobox').mouseenter()");
 
-    waitForNewCategoryToAppear(categoryListSizeBeforeSubmission);
+    removeInfoboxButton.click();
+    removeConfirmationButton.click();
   }
 
-  private void waitForNewCategoryToAppear(final int categoryListSizeBeforeSubmission) {
-    WebDriverWait wait = new WebDriverWait(driver, ELEMENT_SHOW_UP_TIMEOUT);
-    wait.until(new ExpectedCondition<Boolean>() {
-      public Boolean apply(WebDriver driver) {
-        return categoryList.size() > categoryListSizeBeforeSubmission ? true : false;
+  public boolean checkPortableInfoboxIsNotPresent() {
+    return editorFrame.frameScope(() -> {
+      try {
+        wait.forElementNotPresent(portableInfoboxBy);
+        return true;
+      } catch (TimeoutException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)